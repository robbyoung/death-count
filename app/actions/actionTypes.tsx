export enum ActionType {
    ADD_PLAYTHROUGH = 'Add Playthrough',
    ADD_GAME = 'Add Game',
    ADD_DEATH = 'Add Death',
    ADD_DEATH_DETAIL = 'Add Death Detail',
    COMPLETE_DEATH = 'Complete Death',
    ADD_OPTION = 'Add Option',
    ADD_OPTION_SET = 'Add Option Set',
    SELECT_GAME = 'Select Game',
    LOAD_STATE = 'Load State',
    COMPLETE_GAME = 'Complete Game',
    SELECT_PLAYTHROUGH = 'Select Playthrough',
    SELECT_OPTION_SET = 'Select Option Set',
    DELETE_PLAYTHROUGH = 'Delete Playthrough',
}
