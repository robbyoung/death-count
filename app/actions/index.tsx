export * from './actionTypes';
export * from './addPlaythrough';
export * from './addDeath';
export * from './addGame';
export * from './addDeathDetail';
export * from './completeDeath';