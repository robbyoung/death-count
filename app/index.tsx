import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Screens } from './screens';
import Home from './containers/home';
import NewDeath from './containers/newDeath';
import GamesList from './containers/gamesList';
import NewGame from './containers/newGame';
import Loading from './containers/loading';
import Startup from './containers/startup';
import PlaythroughsList from './containers/playthroughsList';

const MainNavigator = createStackNavigator(
	{
		Loading: { screen: Loading },
		Home: { screen: Home },
		NewDeath: { screen: NewDeath },
		GamesList: { screen: GamesList },
		NewGame: { screen: NewGame },
		Startup: { screen: Startup },
		PlaythroughsList: { screen: PlaythroughsList }
	},
	{
		initialRouteName: Screens.Loading,
	},
);

export default createAppContainer(MainNavigator);