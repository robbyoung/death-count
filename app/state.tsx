export interface Game {
    id: string;
    name: string;
    options: string;
}

export interface DeathOptions {
    name: string;
    options: string[];
}

export interface Playthrough {
    id: string;
    name: string;
    description: string;
    gameId: string;
}

export interface Death {
    id: string;
    playthroughId: string;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    currentPlaythrough: string;
    currentGame: string;
}