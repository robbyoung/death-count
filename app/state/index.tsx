export * from './state';
export * from './testState';