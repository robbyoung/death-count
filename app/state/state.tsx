export interface Game {
    id: string;
    name: string;
    selected: boolean;
    complete: boolean;
}

export interface Playthrough {
    id: string;
    name: string;
    gameId: string;
    selected: boolean;
}

export interface Death {
    id: string;
    playthroughId: string;
    complete: boolean;
    details: { [id: string] : string };
}

export interface OptionSet {
    id: string;
    gameId: string;
    title: string;
    options: string[];
    selected: boolean;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    optionSets: OptionSet[];
}