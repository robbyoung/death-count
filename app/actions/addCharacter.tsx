import { Death, Character } from '../state';
import { Action } from 'redux';

export interface AddCharacterAction extends Action {
    newCharacter: Character;
}

export function addCharacter(state: Character[], action: AddCharacterAction): Character[] {
    return [
        ...state,
        action.newCharacter,
    ]
}