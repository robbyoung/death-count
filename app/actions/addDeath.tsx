import { Death } from '../state';
import { Action } from 'redux';
import { ActionType } from './actionTypes';

export class AddDeath implements Action {
    type: ActionType = ActionType.ADD_DEATH;

    constructor(public newDeath: Death) {}
}

export function addDeath(state: Death[], action: AddDeath): Death[] {
    return [
        ...state,
        action.newDeath,
    ]
}