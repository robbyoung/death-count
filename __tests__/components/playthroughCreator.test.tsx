import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../app/store';
import { loadStateAction, addPlaythroughAction, updatePlaythroughAction } from '../../app/actions';
import { createTestState, createTestPlaythrough } from '../../app/state';
import PlaythroughCreator from '../../app/containers/playthroughCreator';

const fakeNavigation = {
    goBack: () => undefined,
}
jest.mock('@react-native-community/async-storage', () => ({}));

describe('Playthrough Creator Container', () => {
    it('Renders with the empty initial state', () => {
        const component = renderer.create(<PlaythroughCreator navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Renders with the default incomplete playthrough', () => {
        const state = createTestState(1, 1, 2, 0, 0);
        store.dispatch(loadStateAction(state));
        store.dispatch(addPlaythroughAction());

        const component = renderer.create(<PlaythroughCreator navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Renders with an edited incomplete playthrough', () => {
        const state = createTestState(2, 1, 2, 0, 0);
        store.dispatch(loadStateAction(state));
        store.dispatch(addPlaythroughAction());
        store.dispatch(updatePlaythroughAction({
            ...createTestPlaythrough(false),
            name: 'New name',
            gameId: 'g1',
        }))

        const component = renderer.create(<PlaythroughCreator navigation={fakeNavigation as any} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
