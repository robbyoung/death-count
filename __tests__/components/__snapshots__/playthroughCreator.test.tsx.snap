// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playthrough Creator Container Renders with an edited incomplete playthrough 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": "5%",
        "marginRight": "5%",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      Playthrough Name
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "color": "#000000",
          "fontSize": 18,
          "padding": 10,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value="New name"
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": "5%",
        "marginRight": "5%",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      Game
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "width": "100%",
        }
      }
    >
      <View>
        <RCTPicker
          items={
            Array [
              Object {
                "label": "Game 0",
                "textColor": undefined,
                "value": "g0",
              },
              Object {
                "label": "Game 1",
                "textColor": undefined,
                "value": "g1",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={1}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 30,
          "height": 50,
          "marginLeft": "5%",
          "marginRight": "5%",
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "90%",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`Playthrough Creator Container Renders with the default incomplete playthrough 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": "5%",
        "marginRight": "5%",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      Playthrough Name
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "color": "#000000",
          "fontSize": 18,
          "padding": 10,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": "5%",
        "marginRight": "5%",
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      Game
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "width": "100%",
        }
      }
    >
      <View>
        <RCTPicker
          items={
            Array [
              Object {
                "label": "None Selected",
                "textColor": undefined,
                "value": undefined,
              },
              Object {
                "label": "Game 0",
                "textColor": undefined,
                "value": "g0",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginTop": 10,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 30,
          "height": 50,
          "marginLeft": "5%",
          "marginRight": "5%",
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "90%",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`Playthrough Creator Container Renders with the empty initial state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#592a29",
      "height": "100%",
    }
  }
/>
`;
